doc/api.yml

```diff
@@ -0,0 +1,783 @@
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+      responses:
+        '201':
+          description: The identified user updated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+
+  ## Following resources
+  '/followers':
+    get:
+      tags:
+        - Following
+      summary: Get all followers from user connected.
+      operationId: getFollowers
+      parameters:
+        - name: page
+          in: query
+          required: true
+          schema:
+            $ref: '#/components/schemas/Page'
+        - name: page_size
+          in: query
+          required: true
+          schema:
+            $ref: '#/components/schemas/PageSize'
+        - name: follower_keyword
+          in: query
+          required: false
+          description: Filter follower by first name or lastname.
+          schema:
+            type: string
+      responses:
+        '200':
+          description: List of followers from user connected.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/User'
+        '400':
+          $ref: '#/components/responses/400'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+
+  '/follow/{user_id}':
+    post:
+      tags:
+        - Following
+      summary: Follow user by identifier.
+      operationId: followUserById
+      parameters:
+        - name: user_id
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: follow_type
+          in: query
+          required: true
+          schema:
+            $ref: '#/components/schemas/FollowType'
+      responses:
+        '201':
+          description: The followed user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        '400':
+          $ref: '#/components/responses/400'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '429':
+          $ref: '#/components/responses/429'
+        '500':
+          $ref: '#/components/responses/500'
+
+  ## Posting resources
+  '/posts':
+    get:
+      tags:
+        - Posting
+      summary: Get all posts.
+      operationId: getPosts
+      parameters:
+        - name: page
+          in: query
+          required: true
+          schema:
+            $ref: '#/components/schemas/Page'
+        - name: page_size
+          in: query
+          required: true
+          schema:
+            $ref: '#/components/schemas/PageSize'
+        - name: post_type
```

Comment:
category => topic

we want to be able to filter a post by sending a topic list 
(like tags)